include/s2e/S2EExecutionStateMemory.h

```diff
@@ -71,21 +66,71 @@ class S2EExecutionStateMemory {
         return (uintptr_t) m_dirtyMask->getConcreteStore(false) - s_dirtyMask->address;
     }
 
+    /// Read/write from physical memory, concretizing if necessary on reads.
+    /// Note: this function accepts host address. Used by softmmu code.
+    void transferRam(struct CPUTLBRAMEntry *te, uint64_t hostAddress, void *buf, uint64_t size, bool isWrite,
+                     bool exitOnSymbolicRead, bool isSymbolic);
+
     klee::ObjectPair getMemoryObject(uint64_t address, AddressType addressType = VirtualAddress) const;
 
     ////////////////////////////////////////////////////////////
     // The APIs below may be used by plugins
     ////////////////////////////////////////////////////////////
 
-    /** Virtual address translation (debug mode). Returns -1 on failure. */
+    ///
+    /// \brief getPhysicalAddress returns the guest physical address for the given virtual address
+    /// \param virtualAddress the virtual address to translate
+    /// \return returns the physical address (or -1 in case of error)
+    ///
     uint64_t getPhysicalAddress(uint64_t virtualAddress) const;
 
-    /** Address translation (debug mode). Returns host address or -1 on failure */
+    ///
+    /// \brief getHostAddress translates the given address to the host address
+    /// \param address the address to translate
+    /// \param addressType the type of address specified
+    /// \return the host address (or -1 in case of failure)
+    ///
     uint64_t getHostAddress(uint64_t address, AddressType addressType = VirtualAddress) const;
 
     /** Read memory to buffer, concretize if necessary */
+
+    ///
+    /// \brief read reads data from memory
```

Comment:
I don’t think you need to include the method name at the start of each comment, it reads a bit funny. It’s probably enough to say “\brief Read data from memory” because we know what method is being documented.